modules/serverless-eventing/pages/service-mesh/eventing-service-mesh-mt-channel-based-broker-authorization.adoc

```diff
@@ -0,0 +1,328 @@
++
+[source,yaml]
+----
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: allow-authz-tests-mt-channel-based-broker
+  namespace: knative-eventing
+spec:
+  action: ALLOW
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: "broker-ingress" <2>
+  rules:
+    - from: <1>
+        - source:
+            namespaces: [ "authz-tests" ]
+      to: <2>
+        - operation:
+            methods: [ "POST" ]
+            paths: [ "/authz-tests/*" ] <3>
+----
+<1> Allow workloads in the `authz-tests` namespace
+<2> To post events to Knative brokers in the `authz-tests` namespace.
+<3> Knative Broker with class MTChannelBasedBroker accepts events on HTTP path following the pattern: `/<broker-namespace>/<broker-name>`.
+
+. Apply the `AuthorizationPolicy` resource:
++
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
+
+. Create a `AuthorizationPolicy` in the `knative-eventing` namespace to allow `mt-broker-ingress` in
+the `knative-eventing` namespace to post events to the `InMemoryChannel` dispatcher:
++
+[source,yaml]
+----
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: allow-mt-channel-based-broker-ingress-to-imc-dispatcher
+  namespace: knative-eventing
+spec:
+  action: ALLOW
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: "imc-dispatcher" <2>
+  rules:
+    - from: <1>
+        - source:
+            namespaces: [ "knative-eventing" ]
+            principals: [ "cluster.local/ns/knative-eventing/sa/mt-broker-ingress" ]
+      to: <2>
+        - operation:
+            methods: [ "POST" ]
+----
+<1> Allow `mt-broker-ingress` in the `knative-eventing` namespace.
+<2> To post events to the `InMemoryChannel` dispatcher.
+
+. Apply the `AuthorizationPolicy` resource:
++
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
+
+. Create a `AuthorizationPolicy` in the `knative-eventing` namespace to allow `imc-dispatcher` in
+the `knative-eventing` namespace to post events to the `mt-broker-filter`:
++
+[source,yaml]
+----
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: allow-imc-dispatcher-to-mt-channel-based-broker-filter
+  namespace: knative-eventing
+spec:
+  action: ALLOW
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: "broker-filter" <2>
+  rules:
+    - from: <1>
+        - source:
+            namespaces: [ "knative-eventing" ]
+            principals: [ "cluster.local/ns/knative-eventing/sa/imc-dispatcher" ]
+      to: <2>
+        - operation:
+            methods: [ "POST" ]
+----
+<1> Allow `imc-dispatcher` in the `knative-eventing` namespace.
+<2> To post events to the `mt-broker-filter`.
+
+. Apply the `AuthorizationPolicy` resource:
++
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
+
+.Verification
+
+You can verify that the events were sent to the Knative event sink by looking at the message dumper function logs.
```

Comment:
I really like the E2E example, and the detailed configuration example here for patching/configuring all the components needed, in order to make this work